app/views/pair_requests/_form.html.erb

```diff
@@ -1,23 +1,32 @@
 <%= turbo_frame_tag "new_request_form" do %>
-  <%= form_with(model: pair_request, class: 'p-4 w-full mx-auto border-2 border-primary rounded-md flex flex-col') do |form| %>
-    <div id="form-container" class='flex flex-col space-y-4 xl:space-y-0 xl:max-w-full xl:inline-flex xl:justify-around xl:space-x-8 xl:flex-row'>
-      <div class="flex justify-between space-x-4 align-middle">
-        <%= form.label :invitee_id, class: "label label-text" %>
-        <%= form.collection_select :invitee_id, User.excluding(current_user), :id, :full_name, {}, { class: 'select select-primary' } %>
+  <%= form_with(model: pair_request, class: 'p-4 mx-auto border-2 border-primary rounded-md flex flex-col') do |form| %>
+    <div id="form-container" class='flex flex-col space-y-4 xl:space-y-0 xl:max-w-full xl:inline-flex xl:justify-around xl:space-x-4 xl:flex-row object-contain' data-controller="pair-requests-form">  
+      <div class="flex justify-between space-x-4 align-middle items-center ">
+         <%= form.label :invitee_id, class: "label label-text" %>
+         <%= form.select :invitee_id, options_for_select(User.excluding(current_user).map{|invitee| [invitee.full_name, invitee.id, { "data-timezone-identifier" => ActiveSupport::TimeZone.new(invitee.time_zone).tzinfo.identifier, "data-timezone-display-name" => invitee.time_zone, "data-usertz" => current_user.time_zone}]}), {:include_blank => "Please select"}, { class: 'select select-primary', data: {action: "change->pair-requests-form#onClick",  "pair-requests-form-target" => "inviteeInfo"}} %>
```

Comment:
I don't know whether this suggestion is blocking, will defer to @toyhammered . 
This seems like a lot of logic being put in the view, and I think the instantiation of the `ActiveSupport::TimeZone` instance might be the thing pushes it over for me.

I wonder whether this type of logic might be better located in a user decorator, like `Invitee`, and then you can access those invitees using a `@invitees` instance variable?
If you go this route, you'd also need to make sure that a `delegators` folder is slurped up by the application so you'd be able to call it.
e.g.:
```ruby
# app/delegators/invitee.rb
class Invitee < DelegateClass(User)
  def time_zone_identifier
    ActiveSupport::TimeZone.new(time_zone).tzinfo.identifier
  end
end
```
then in the controller, something like
```ruby
# app/controllers/pair_requests_controller.rb
@invitees = User.excluding(current_user).map { |user| Invitee.new(user) }
```

Reply:
Oh yeah you're totally right about all the logic here. I still need to implement Josh's suggestion on cleaning it up and I like your suggestion to break it up like that. Thank you Ariel!

Reply:
Hmm, would it be an ok approach to make that delegator method into an instance method for the class? I'm realizing I might need to do the same AS::TZ instantiation for the current user as well.

Reply:
I think I was still just wrap the current_user in an `Inviter` delagator where you need it, since the `time_zone_identifier` behavior is context specific to when invitations are necessary, but that's just me. Would be curious to hear what @toyhammered says.

Reply:
Ah ok so in that scenario, in the controller id do something like: `@current_user = Invitee.new(current_user)`

Reply:
I guess the question is -- when do you need the current user to have access to the `time_zone_identifier`? how often and for what behavior?

Reply:
That's a very fair question since a lot of the time, we'd only really need to use `current_user.time_zone` when we'd want to display a simple time zone and this identifier is only getting used for the datetime_field conversion.